doc/mgmt/SONiC_OC_Interface_Enhancements_HLD.md

```diff
@@ -0,0 +1,373 @@
+* By default, the VLAN's state in kernel is down and changes to up only when first port/LAG in the VLAN is oper up or autostate is disabled and admin state is up.
+
+#### Loopback
+
+* Add support for admin state configuration via KLISH CLI option-"[no] shutdown", REST or gNMI. 
+* By default, admin state & oper state is up. 
+
+## 1.1 Target Deployment Use Cases
+Enable/disable VLAN and Loopback interfaces via gNMI, REST or KLISH CLI.
+
+## 1.2 Requirements
+
+### 1.2.1 Functional requirements
+1. Provide management framework capability to handle admin state configuration for VLAN & Loopback interfaces.
+
+### 1.2.2 Configuration and Management Requirements
+
+#### 1.2.2.1 KLISH requirements
+
+1. To administratively bring the VLAN and Loopback interface down, use "shutdown" command.
+2. To return the interface to its default admin state, use "no shutdown".
+3. The above configurations should be displayed as part of a show command.  
+
+## 1.3 Design Overview
+
+### 1.3.1 Basic Approach
+#### 1.3.1.1 VLAN
+
+1. The VLAN table in the CONFIG_DB supports admin state up/down parameter.
+2. The VlanMgr to be enhanced to populate the APP_DB with the admin state up/down.
+3. The orchagent to be enhanced to update VLAN's oper state based on admin state. 
+   The contributors to the the operational status are as follows: 
+   - Admin state.
+   - Autostate.
+   - Physical Port members.
+   - LAG members.
+4. When the configuration changes from admin state up to down the operational status will change to operationally down. If the state were to be already operationally down then it will be a no-op. 
+5. When the configuration changes from admin down to up the operational status will change to operationally up based on autostate and members operational state.  
+6. For the VLAN state in kernel, the existing mechanism to notify the VlanMgr about the operational status will be reused. 
+
+#### 1.3.1.2 LOOPBACK
+1. Update LOOPBACK table in the CONFIG_DB to support admin state up/down parameter.
+2. The LOOPBACK_TABLE in APP_DB already supports admin state up/down parameter.
+3. The oper status will be controlled by the admin status. 
+4. When the configuration changes from admin state up to down the operational status will change to operationally down.  
+5. When the configuration changes from admin down to up the operational status will change to operationally up.   
+
+### 1.3.2 Container
+* **swss container** : 
+  * VlanMgr and Orchagent changes to also consider admin status while setting oper status.
+* **management-framework** :
+  * CLI XML & Jinja-template changes to support CLI-style configurations and relevant show commands.
+  
+### 1.3.3 SAI Overview
+N.A
+
+# 2 Functionality
+As Described in section 1.2
+
+# 3 Design
+## 3.1 Overview
+As described in section 1.3
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+**VLAN table**
+* No change.
+
+**LOOPBACK table**
+* Producer: Mangement framework
```

Comment:
Please change the producer to be config-DB.

Reply:
Changed to  - Producer: Mangement framework/config_db.json/Click command